### PR TITLE
Add MariaDB migration testing

### PR DESCRIPTION
**Pull request by strickvl**
Adds MariaDB DB migration testing.

https://github.com/zenml-io/zenml/actions/runs/7452053752/job/20274610586 shows the test running through without error.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

## Summary of changes

- **New Features**
  - Added a new workflow to test MariaDB database migrations upon pull requests.

- **Enhancements**
  - Improved the migration testing script to include support for both MariaDB and MySQL databases.

- **Documentation**
  - Updated workflow documentation to reflect the new testing process and triggers.

### CODE REVIEW

**Comment by strickvl** [comment]
@coderabbitai review

File: scripts/test-migrations.sh [hunk 1]
```diff
@@ -41,9 +41,14 @@ if [ "$1" == "mysql" ]; then
...
     docker run --name mysql -d -p 3306:3306 -e MYSQL_ROOT_PASSWORD=password mysql:latest
-
     # mysql takes a while to start up
     sleep 30
+elif [ "$1" == "mariadb" ]; then
+    echo "===== Testing MariaDB ====="
+    # run a mariadb instance in docker
+    docker run --name mariadb -d -p 3306:3306 -e MYSQL_ROOT_PASSWORD=password mariadb:10.6
+    # mariadb takes a while to start up
+    sleep 30
```
[the remark this answers is not in the document]
Comment (strickvl) [reply scripts/test-migrations.sh:51]: Can you make a code suggestion to handle that parameterization?

File: .github/workflows/testing_this_now.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (stefannica) [inline .github/workflows/testing_this_now.yml:1]: weird choice for a file name :smile_cat:

**Review by stefannica — APPROVED** [review]
looks good

File: .github/workflows/testing_this_now.yml (hunk 2 above)
Comment (strickvl) [reply .github/workflows/testing_this_now.yml:1]: haha just so I can run it quickly on the CI... will delete this before it's merged.

**Comment by strickvl** [comment]
@coderabbitai review

File: scripts/test-migrations-mariadb.sh [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+function run_tests_for_version() {
+    set -e  # Exit immediately if a command exits with a non-zero status
+    local VERSION=$1
+
+    echo "===== Testing version $VERSION ====="
+    # Initialize zenml with the appropriate template
+    # hardcoded to 0.43.0 since this is the latest template-starter repo
+    # release tag
+    copier copy -l --trust -r release/0.43.0 https://github.com/zenml-io/template-starter.git test_starter
```
Comment (avishniakov) [inline scripts/test-migrations-mariadb.sh:14]: Specifically this part can be handled via `zenml init ...` instead

**Review by avishniakov — CHANGES_REQUESTED** [review]
Same observations applies to the existing MySQL code

**Comment by strickvl** [comment]
@avishniakov fixed as per your changes and all ran without error here https://github.com/zenml-io/zenml/actions/runs/7463701141

**Review by avishniakov — APPROVED** [review]

**Pull request merged**